### PR TITLE
Oracle JDK 21 support

### PR DESCRIPTION
**Pull request by jdubois**
Add test and documentation for Oracle JDK 21 support.

There will be more to do when the other distributions support it, but it's the first at the moment.

**Check list:**
- [x] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by fniephaus** [comment]
Thanks for updating! Please note that this is Oracle JDK and not OpenJDK.

**Comment by jdubois** [comment]
You’re right @fniephaus ! I’ve been typing OpenJDK too much at work! I updated the title and description.

**Comment by rjeberhard** [comment]
What's the status of this PR? I'd really appreciate being able to use JDK 21 in actions. Thanks!

**Comment by IvanZosimov** [comment]
Hi, @rjeberhard 👋 We'll take a look at the PR, but even without it, you already can install Oracle JDK 21 with setup-java.

File: README.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ steps:
 
 #### Supported version syntax
 The `java-version` input supports an exact version or a version range using [SemVer](https://semver.org/) notation:
-- major versions: `8`, `11`, `16`, `17`
+- major versions: `8`, `11`, `16`, `17`, `21`
 - more specific versions: `17.0`, `11.0`, `11.0.4`, `8.0.232`, `8.0.282+8`
```
Comment (yeikel) [inline README.md:92]: Does this also add support for specific versions? ie : `21.0.1+12` or will this be included in a subsequent change?

Comment (jdubois) [reply README.md:92]: That's only the documentation, but yes more generally the minor and patch versions will work

**Review by asolntsev — APPROVED** [review]

**Comment by asolntsev** [comment]
@jdubois @sormuras Is there a plan to release this? Java 21 was released 2 months ago already..

**Comment by jdubois** [comment]
@asolntsev I'm sorry but I just contributed this small PR and I'm not a member of the project, I have no idea when they'll do a release

**Comment by fniephaus** [comment]
Oracle JDK 21 should already be working fine. This PR just adds documentation and a test for it. Just specify java-version 21 or 21.0.1 or something like that and you should be good to go. Otherwise, please notify us here 🙂

**Review by IvanZosimov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**